ledfx/effects/audio.py

```diff
@@ -792,10 +794,14 @@ def activate(self, channel):
             )
 
         self.audio = self._ledfx.audio
-        self._ledfx.audio.subscribe(self._audio_data_updated)
+        self.audio.subscribe(self._audio_data_updated)
```

Comment:
This leaves me uncomfortable as a call to activate on a deactivated instance would not re-establish subscription to audio.

Resource management should be paired between activate / deactivate. 

I can't point yet at a scenario that would trigger this, but as a future dev, it would be an expectation that it could be used in such a way.

Reply:
A call to deactivate and then activate creates a new instance of AudioReactiveEffect (in fact its new instances all the way down the class tree) so this shouldn't be a problem - if we look at the base Effect class, its init method is how we get the ledfx object and other important things like config and the like.

I've verified this by looking at the ID of the audioreactiveeffect class and it changes for every effect, which groks with my understanding of how its structured.

I don't *think* this will cause any issues unless we start persisting base classes for effects, which might make sense from a resource utilisation POV but given the associated complexity and our lack of bottlenecks at the current time, shouldn't be a thing we do/focus on.

Reply:
Lets go with this then, but keep an eye open dor the unexpected. There are some funky audio related crashes further down the list that this "may" help with...

Reply:
> Resource management should be paired between activate / deactivate

I agree this is the commonly expected behavior.

If you want to proceed with breaking the symmetry between activate/deactivate, I suggest adding a guard to `activate` and `raise Exception("Effect cannot be reactivated once deactivated")`. This will prevent future developers unnecessarily wasting time.

It seems to me that initializing the audio class attribute should be sufficient, though.